aws/resource_aws_codecommit_approval_rule_association.go

```diff
@@ -0,0 +1,148 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsCodeCommitApprovalRuleAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeCommitApprovalRuleAssociationCreate,
+		Update: resourceAwsCodeCommitApprovalRuleAssociationUpdate,
+		Read:   resourceAwsCodeCommitApprovalRuleAssociationRead,
+		Delete: resourceAwsCodeCommitApprovalRuleAssociationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"template_name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"repository_names": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
```

Comment:
i think we should stick to api and have this be a single value and users can use `for_each` if needed